chainexchange/pubsub.go

```diff
@@ -270,6 +274,9 @@ func (p *PubSubChainExchange) cacheAsWantedChain(ctx context.Context, cmsg Messa
 				fullChain: &cmsg.Chain,
 				offset:    offset,
 			})
+			if p.listener != nil {
+				p.listener.NotifyChainDiscovered(ctx, key, cmsg.Instance, prefix)
```

Comment:
IDK if the notification under lock is the best idea. It might be better to move it outside the lock.